File: PRACTICA02/JAVA - CLASE FRACCION/TestFraccion.java
public class TestFraccion {

public static void main(String [] args) {

Fraccion f1 = new Fraccion(1, 4);
Fraccion f2 = new Fraccion(4, 3);

System.out.println("f1 = " + f1);
System.out.println("f2 = " + f2);

System.out.println("suma = " + f1.suma(f2));
System.out.println("resta = " + f1.resta(f2));
System.out.println("multiplica = " + f1.multiplica(f2));
System.out.println("divide = " + f1.divide(f2));
System.out.println("decimal = " + f1.convertirADecimal());
System.out.println("inverso = " + f1.esInverso(f2));

Fraccion f3 = Fraccion.parseFraccion("-2/3");
System.out.println("parse = " + f3);

System.out.println("simplificada = " + new Fraccion(2,8).simplifica());
}
}

File: PRACTICA02/PYTHON - CLASE FRACCION/testfraccion.py
from fraccion import Fraccion

f1 = Fraccion (1,4)
f2 = Fraccion (4,3)

print("f1 =", f1)
print("f2 =", f2)

print("suma =", f1 + f2)
print("resta =", f1 - f2)
print("multiplica =", f1 * f2)
print("divide =", f1 / f2)
print("decimal =", f1.convertirADecimal())
print("inverso =", f1.esInverso(f2))

f3 = f1.parseFraccion("-2/3")
print("parse =", f3)

print("simplificada =", Fraccion(2,8).simplifica())
